src/app.ts

```diff
@@ -0,0 +1,61 @@
+import * as bodyParser from "body-parser";
+import * as express from "express";
+import { Pool } from "pg";
+
+import { log } from "./utils/logger";
+
+// Create Express server
+const app = express();
+
+const timeout = 5000;
+
+const postgres = new Pool({
+  connectionString: "postgres://postgres:password@localhost/postgres"
+});
+
+function createConnection(db: Pool): Promise<void> {
+  return new Promise((resolve, reject) => {
+    db.connect((errorConnect, _2, done) => {
+      if (errorConnect) {
+        return reject(errorConnect);
+      }
+      resolve();
+      done();
+    });
+  });
+}
+
+// tslint:disable-next-line:no-let
+let retries = 10;
+function tryToConnect(): Promise<void> {
+  return new Promise(async (resolve, reject) => {
+    while (retries) {
+      try {
+        log.info("Attempting to connect to database...");
+        await createConnection(postgres);
```

Comment:
I think you don't need `tryToConnect` at all, if `createConnection` fails just exit the process, the supervisor (docker compose or kubernetes) will relaunch the server with backoff